src/peripheral/dma/ccr.rs

```diff
@@ -0,0 +1,397 @@
+/*
+* Copyright (C) 2017 AltOS-Rust Team
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use super::defs::*;
+
+/// Defines the possible directions that data can be read from.
+#[derive(Copy, Clone, Debug)]
+pub enum DataDirection {
+    /// Data is read from the peripheral.
+    FromPeriph,
+    /// Data is read from memory.
+    FromMem,
+}
+
+/// Defines the possible transfer data sizes of the peripheral and memory.
+#[derive(Copy, Clone, Debug)]
+pub enum PeriphAndMemSize {
+    /// Eight bits.
+    Eight,
+    /// Sixteen bits.
+    Sixteen,
+    /// Thirty-two bits.
+    ThirtyTwo,
+    /// Reserved.
+    Reserved,
```

Comment:
Any reason to have a reserved variant? Seems like it's just reserved by hardware and you shouldn't be able to set it to that value.